sla_per_user_system_test.py

```diff
@@ -116,46 +127,40 @@ def validate_scheduler_runtime(self, start_time, end_time, read_users, expected_
+                                        f'Run time per user: {runtime_per_user}')
 
     @staticmethod
-    def create_auths(entities_list_of_dict):
-        """
-        :param entities_list_of_dict: Expected structure:
-                    [{'user': User(), 'role': Role(), 'service_level': ServiceLevel()},
-                    OR
-                     {'user': User(), 'service_level': ServiceLevel()},
-                    OR
-                     {'role': Role(), 'service_level': ServiceLevel()}
-                    ]
-        """
-        for entity in entities_list_of_dict:
-            service_level = entity.get('service_level')
-            role = entity.get('role')
-            user = entity.get('user')
-            if service_level:
-                service_level.create()
-            if role:
-                role.create()
-                role.attach_service_level(service_level=service_level)
-            if user:
-                user.create()
-                if role:
-                    role.grant_me_to(grant_to=user)
-                else:
-                    user.attach_service_level(service_level=service_level)
+    def validate_ratio(expected_ratio, actual_ratio, msg):
+        if not (expected_ratio and actual_ratio):
+            WorkloadPrioritisationEvent.RatioValidationEvent(
+                message=f'Can\'t compare expected and actual shares ratio. Expected: {expected_ratio}. '
+                        f'Actual: {actual_ratio}', severity=Severity.ERROR).publish()
+
+        elif expected_ratio <= actual_ratio:
+            WorkloadPrioritisationEvent.RatioValidationEvent(
+                message=f'{msg}. Actual ratio ({actual_ratio}) is as expected (more or equal then expected ratio '
+                        f'{expected_ratio})',
+                severity=Severity.NORMAL).publish()
+        else:
+            WorkloadPrioritisationEvent.RatioValidationEvent(
+                message=f'{msg}. Actual ratio ({actual_ratio}) is less then expected ratio ({expected_ratio})',
+                severity=Severity.ERROR).publish()
 
     def validate_deviation(self, expected_ratio, actual_ratio, msg):
         dev = self.calculate_deviation(expected_ratio, actual_ratio)
-        self.assertIsNotNone(dev, 'Can\'t compare expected and actual shares ratio. Expected: '
-                                  '{expected_ratio}. Actual: {actual_ratio}'
-                             .format(expected_ratio=expected_ratio, actual_ratio=actual_ratio)
-                             )
-        # TODO: formulate error message
-        self.assertTrue(dev <= self.VALID_DEVIATION_PRC, '{msg}. Actual shares ratio ({actual_ratio}) is not '
-                                                         'as expected ({expected_ratio})'
-                        .format(msg=msg, actual_ratio=actual_ratio, expected_ratio=expected_ratio)
-                        )
+        if not dev:
```

Comment:
What if `dev == 0.0`? It's a valid deviation value but it evaluates to `False`. Probably better to replace with `if not None`

Reply:
changed